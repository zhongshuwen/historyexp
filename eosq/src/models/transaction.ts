import { ActionTrace, Action } from "@dfuse/client"

export enum TransactionReceiptStatus {
  EXECUTED = "executed",
  SOFT_FAIL = "soft_fail",
  HARD_FAIL = "hard_fail",
  DELAYED = "delayed",
  EXPIRED = "expired",
  PENDING = "pending",
  CANCELED = "canceled"
}

export interface DeferredOperation {
  transaction_id: string
  action_index: number
  by_transaction_id?: string
  operation: string
  sender: string
  sender_id: string
  payer: string
  published_at: string
  delay_until: string
  expiration_at: string
  related_transactions: string[]
}

export interface TraceLevel {
  index: number
  actionTrace: ActionTrace<any>
  group: number
  level: number
}

/**
 * Represents a transaction as it passes through the P2P channel, not yet
 * executed since not part of a block.
 */
export interface Transaction {
  id: string
  block_num: number
  block_id: string
  producer: string
  ref_block_num: number
  ref_block_prefix: number
  max_net_usage_words: number
  delay_sec: number
  expiration: string
  max_cpu_usage_ms: number
  context_free_actions: Action<any>[]
  actions: Action<any>[]
  transaction_extensions: any[]
  signatures: string[]
  public_keys: string[]
  irreversible: boolean
}

/**
 * Represents a transaction action aggregate to be displayed in the list.
 * This models aggregates is an action with extended information from the
 * transaction itself (id, block_num, date)
 */
export interface TransactionAction extends Action<any> {
  action_num: string
  cfa: boolean
}

export enum RAMOpTypes {
  CREATE_TABLE = "create_table",
  DEFERRED_TRX_ADD = "deferred_trx_add",
  DEFERRED_TRX_CANCEL = "deferred_trx_cancel",
  DEFERRED_TRX_PUSHED = "deferred_trx_pushed",
  DEFERRED_TRX_REMOVED = "deferred_trx_removed",
  DELETEAUTH = "deleteauth",
  LINKAUTH = "linkauth",
  NEWACCOUNT = "newaccount",
  PRIMARY_INDEX_ADD = "primary_index_add",
  PRIMARY_INDEX_REMOVE = "primary_index_remove",
  PRIMARY_INDEX_UPDATE = "primary_index_update",
  PRIMARY_INDEX_UPDATE_ADD_NEW_PAYER = "primary_index_update_add_new_payer",
  PRIMARY_INDEX_UPDATE_REMOVE_OLD_PAYER = "primary_index_update_remove_old_payer",
  REMOVE_TABLE = "remove_table",
  SECONDARY_INDEX_ADD = "secondary_index_add",
  SECONDARY_INDEX_REMOVE = "secondary_index_remove",
  SECONDARY_INDEX_UPDATE_ADD_NEW_PAYER = "secondary_index_update_add_new_payer",
  SECONDARY_INDEX_UPDATE_REMOVE_OLD_PAYER = "secondary_index_update_remove_old_payer",
  SETABI = "setabi",
  SETCODE = "setcode",
  UNLINKAUTH = "unlinkauth",
  UPDATEAUTH_CREATE = "updateauth_create",
  UPDATEAUTH_UPDATE = "updateauth_update"
}

export function computeTransactionTrustPercentage(
  blockNum: number | undefined,
  headBlockNum: number,
  lastIrreversibleBlockNum: number
) {
  if (blockNum === undefined) return 0.0
  if (lastIrreversibleBlockNum >= blockNum) return 1.0

  const blockPassedCount = headBlockNum - blockNum
  if (blockPassedCount >= 360) {
    return 0.9999
  }

  if (blockPassedCount < 4) {
    return blockPassedCount / 4
  }

  if (blockPassedCount === 4) {
    return 0.99
  }

  return (blockPassedCount / 360) * 0.01 + 0.99
}
