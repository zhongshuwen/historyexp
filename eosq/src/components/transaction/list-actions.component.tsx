import * as React from "react"
import { DeferredOperation } from "../../models/transaction"
import { ActionPill } from "../action-pills/action-pill.component"
import { Cell } from "../../atoms/ui-grid/ui-grid.component"
import { DeferredLink } from "../deferred-link/deferred-link"
import { DbOp, RAMOp, Action } from "@dfuse/client"

interface Props {
  actions: Action<any>[]
  deferredOperations?: DeferredOperation[]
  ramops?: RAMOp[]
  dbops?: DbOp[]
  actionIndexes?: number[]
}

export class ListActions extends React.Component<Props> {
  renderItem = (action: Action<any>, index: number) => {
    let operations: DeferredOperation[] = []

    if (this.props.deferredOperations) {
      operations = this.props.deferredOperations.filter(
        (operation) => operation.action_index === index && operation.operation !== "PUSH_CREATE"
      )
    }

    const ramops = (this.props.ramops || []).filter(
      (ramop: RAMOp) => ramop.action_idx === index && ramop.op !== "deferred_trx_removed"
    )
    const dbops = (this.props.dbops || []).filter((dbop: DbOp) => dbop.action_idx === index)

    return (
      <Cell key={index}>
        <Cell>
          <ActionPill ramops={ramops} action={action} dbops={dbops} />
        </Cell>
        <Cell pl="25px">
          {operations.map((operation: DeferredOperation, idx: number) => {
            return (
              <Cell key={idx} mt={[1]}>
                <DeferredLink
                  transactionId={operation.transaction_id}
                  operation={operation.operation}
                  delay={operation.delay_until}
                />
              </Cell>
            )
          })}
        </Cell>
      </Cell>
    )
  }

  renderItems = () => {
    return this.props.actions.map((action: Action<any>, index: number) => {
      return this.renderItem(action, index)
    })
  }

  render() {
    return <Cell mb={[2]}>{this.renderItems()}</Cell>
  }
}
