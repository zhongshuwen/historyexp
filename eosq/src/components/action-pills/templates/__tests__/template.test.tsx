import * as React from "react"
import { Action } from "@dfuse/client"
import { shallowWithTheme } from "../../../../tests/renderers"
import { getActionMock } from "../../../../__mocks__/transaction.mock"
import { BuyRamBytesPillComponent } from "../system/buy-ram-bytes-pill.component"
import { BuyRamPillComponent } from "../system/buy-ram-pill.component"
import { ClaimRewardsPillComponent } from "../system/claim-rewards-pill.component"
import { DelegateBandwidthPillComponent } from "../system/delegate-bandwidth-pill.component"
import { IssuePillComponent } from "../system/issue-pill.component"
import { LinkAuthPillComponent } from "../system/linkauth-pill.component"
import { NewAccountPillComponent } from "../system/newaccount-pill.component"
import { RefundPillComponent } from "../system/refund-pill.component"
import { RegProxyPillComponent } from "../system/regproxy-pill.component"
import { TransferPillComponent } from "../transfer-pill.component"
import { UnDelegateBandwidthPillComponent } from "../system/undelegate-bandwidth-pill.component"
import { UpdateAuthPillComponent } from "../system/updateauth-pill.component"
import { VotePillComponent } from "../system/vote-pill.component"
import { GenericPillComponent } from "../generic-pill.component"
import { SetcodePillComponent } from "../system/setcode-pill.component"

const headerAndTitleOptions = {
  header: {
    color: "#33333",
    text: "text",
    hoverTitle: "hovetTitle"
  },
  title: "title"
}

describe("Templates", () => {
  describe("BuyRamBytesPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          payer: "payer",
          receiver: "receiver",
          bytes: 150
        }
      })

      expect(shallowWithTheme(renderComponent(BuyRamBytesPillComponent, action))).toMatchSnapshot()
    })
  })

  describe("BuyRamPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          payer: "payer",
          receiver: "receiver",
          quantity: "20.0000 EOS"
        }
      })

      expect(shallowWithTheme(renderComponent(BuyRamPillComponent, action))).toMatchSnapshot()
    })
  })

  describe("ClaimRewardsPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          owner: "owner"
        }
      })

      expect(shallowWithTheme(renderComponent(ClaimRewardsPillComponent, action))).toMatchSnapshot()
    })
  })


  describe("DelegateBandwidthPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          from: "from",
          receiver: "receiver",
          stake_cpu_quantity: "20.0000 EOS",
          stake_net_quantity: "10.0000 EOS"
        }
      })

      expect(
        shallowWithTheme(renderComponent(DelegateBandwidthPillComponent, action))
      ).toMatchSnapshot()
    })
  })

  describe("GenericPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          random_field: "random_field",
          other_field: "other_field"
        }
      })

      expect(shallowWithTheme(renderComponent(GenericPillComponent, action))).toMatchSnapshot()
    })

    it("should render properly when no data but hex data present", () => {
      const action = getActionMock({
        data: undefined
      })

      expect(shallowWithTheme(renderComponent(GenericPillComponent, action))).toMatchSnapshot()
    })

    it("should render properly when no data and no hex data present", () => {
      const action = getActionMock({
        data: undefined
      })

      delete action.hex_data

      expect(shallowWithTheme(renderComponent(GenericPillComponent, action))).toMatchSnapshot()
    })
  })

  describe("IssuePillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          to: "eoscanadacom",
          quantity: "10.0000 EOS"
        }
      })

      expect(shallowWithTheme(renderComponent(IssuePillComponent, action))).toMatchSnapshot()
    })
  })

  describe("LinkAuthPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          account: "eoscanadacom",
          requirement: "requirement",
          type: "type"
        }
      })

      expect(shallowWithTheme(renderComponent(LinkAuthPillComponent, action))).toMatchSnapshot()
    })
  })

  describe("NewAccountPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          creator: "eoscanadacom",
          name: "newaccount",
          owner: "newaccount",
          active: true
        }
      })

      expect(shallowWithTheme(renderComponent(NewAccountPillComponent, action))).toMatchSnapshot()
    })
  })

  describe("RefundPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          owner: "newaccount"
        }
      })

      expect(shallowWithTheme(renderComponent(RefundPillComponent, action))).toMatchSnapshot()
    })
  })

  describe("RegProxyPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          isproxy: false,
          proxy: "testing"
        }
      })

      expect(shallowWithTheme(renderComponent(RegProxyPillComponent, action))).toMatchSnapshot()
    })
  })


  describe("BaseSetcodePillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          code: "code",
          account: "account"
        }
      })

      expect(shallowWithTheme(renderComponent(SetcodePillComponent, action))).toMatchSnapshot()
    })
  })

  describe("TransferPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          from: "from",
          to: "to",
          quantity: "2.000 EOS"
        }
      })

      expect(shallowWithTheme(renderComponent(TransferPillComponent, action))).toMatchSnapshot()
    })
  })

  describe("UnDelegateBandwidthPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          from: "from",
          unstake_net_quantity: "1.000 EOS",
          unstake_cpu_quantity: "2.000 EOS"
        }
      })

      expect(
        shallowWithTheme(renderComponent(UnDelegateBandwidthPillComponent, action))
      ).toMatchSnapshot()
    })
  })

  describe("UpdateAuthPillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          auth: "auth",
          parent: "parent",
          permission: "permission"
        }
      })

      expect(shallowWithTheme(renderComponent(UpdateAuthPillComponent, action))).toMatchSnapshot()
    })
  })

  describe("VotePillComponent", () => {
    it("should render properly", () => {
      const action = getActionMock({
        data: {
          voter: "voter",
          producers: ["producer"]
        }
      })

      expect(shallowWithTheme(renderComponent(VotePillComponent, action))).toMatchSnapshot()
    })
  })
})

function renderComponent(Component: any, action: Action<any>) {
  return <Component action={action} headerAndTitleOptions={headerAndTitleOptions} />
}
