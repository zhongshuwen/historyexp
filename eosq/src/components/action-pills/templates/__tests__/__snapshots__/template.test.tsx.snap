// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates BaseSetcodePillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Styled(div)
        fontSize={
          Array [
            1,
          ]
        }
      >
        SHA256[ 
        <b>
          7941cb07924fdc7b710e11d98d82850e89566e1c3cb980517ffe4b430f86dfd5
        </b>
         ] set for account
         
        <MonospaceTextLink
          fontSize={
            Array [
              1,
            ]
          }
          to="/account/account"
        >
          account
        </MonospaceTextLink>
      </Styled(div)>
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;


exports[`Templates BuyRamBytesPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minHeight="26px"
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "payer",
              "type": "accountLink",
              "value": "payer",
            },
            Object {
              "name": "bytes",
              "type": "bold",
              "value": "150 B",
            },
            Object {
              "name": "receiver",
              "type": "accountLink",
              "value": "receiver",
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.buyrambytes.summary"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates BuyRamPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minHeight="26px"
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "payer",
              "type": "accountLink",
              "value": "payer",
            },
            Object {
              "name": "amountEOS",
              "type": "bold",
              "value": "20.0000 EOS",
            },
            Object {
              "name": "receiver",
              "type": "accountLink",
              "value": "receiver",
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.buyram.summary"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates ClaimRewardsPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <KeyValueFormatEllipsis
        content="owner: owner"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates DelegateBandwidthPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minHeight="26px"
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "from",
              "type": "accountLink",
              "value": "from",
            },
            Object {
              "name": "amountCPU",
              "type": "bold",
              "value": "20.0000 EOS",
            },
            Object {
              "name": "amountNET",
              "type": "bold",
              "value": "10.0000 EOS",
            },
            Object {
              "name": "to",
              "type": "accountLink",
              "value": "receiver",
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.delegatebw.summary"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates GenericPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <KeyValueFormatEllipsis
        content="random_field: random_field other_field: other_field"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates GenericPillComponent should render properly when no data and no hex data present 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <KeyValueFormatEllipsis
        content=""
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates GenericPillComponent should render properly when no data but hex data present 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <KeyValueFormatEllipsis
        content="hex_data"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates IssuePillComponent should render properly 1`] = `
<Pill
  content={
    <TransferBox
      amount="10.0000 EOS"
      from="zswhq"
      to="eoscanadacom"
    />
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates LinkAuthPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minHeight="26px"
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "account",
              "type": "accountLink",
              "value": "eoscanadacom",
            },
            Object {
              "name": "requirement",
              "type": "bold",
              "value": "requirement",
            },
            Object {
              "name": "type",
              "type": "bold",
              "value": "type",
            },
            Object {
              "name": "code",
              "type": "accountLink",
              "value": undefined,
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.linkauth.summary"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates NewAccountPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minHeight="26px"
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "creator",
              "type": "accountLink",
              "value": "eoscanadacom",
            },
            Object {
              "name": "name",
              "type": "accountLink",
              "value": "newaccount",
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.newaccount.summary"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates RefundPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <KeyValueFormatEllipsis
        content="owner: newaccount"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates RegProxyPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "account",
              "type": "accountLink",
              "value": "testing",
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.regProxy.summaryUnreg"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates TransferPillComponent should render properly 1`] = `
<Pill
  content={
    <TransferBox
      amount="2.000 EOS"
      from="from"
      to="to"
    />
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="traceActionSendBackground"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates UnDelegateBandwidthPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minHeight="26px"
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "from",
              "type": "accountLink",
              "value": "from",
            },
            Object {
              "name": "amountCPU",
              "type": "bold",
              "value": "2.000 EOS",
            },
            Object {
              "name": "amountNET",
              "type": "bold",
              "value": "1.000 EOS",
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.undelegatebw.summary"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates UpdateAuthPillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minHeight="26px"
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "account",
              "type": "accountLink",
              "value": undefined,
            },
            Object {
              "name": "permission",
              "type": "bold",
              "value": "permission",
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.updateauth.summary"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;

exports[`Templates VotePillComponent should render properly 1`] = `
<Pill
  content={
    <Styled(div)
      alignItems="center"
      display="flex"
      fontSize={
        Array [
          1,
        ]
      }
      minHeight="26px"
      minWidth="10px"
      mx={
        Array [
          2,
        ]
      }
      position="relative"
    >
      <Translate(BaseFormattedText)
        fields={
          Array [
            Object {
              "name": "account",
              "type": "accountLink",
              "value": "voter",
            },
          ]
        }
        fontSize={
          Array [
            1,
          ]
        }
        i18nKey="pillTemplates.voteForBp.summary"
      />
    </Styled(div)>
  }
  expandButtonBgColor="#8d939a"
  expandButtonColor="#fff"
  headerBgColor="#8d939a"
  headerHoverTitle="hovetTitle"
  headerText="text"
  pill2={
    <Styled(div)
      bg="#eaeff5"
      display="flex"
      position="relative"
    >
      <Styled(div)
        alignSelf="center"
        color="text"
        fontSize={
          Array [
            1,
          ]
        }
        px={
          Array [
            2,
          ]
        }
      >
        title
      </Styled(div)>
    </Styled(div)>
  }
  renderExpandedContent={[Function]}
  renderInfo={[Function]}
/>
`;
